### PR TITLE
dev/core#2487 Update defaults for civicrm_contribution_recur table

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#2487 Update defaults for civicrm_contribution_recur table

Per https://lab.civicrm.org/dev/core/-/issues/2487 this brings the api behaviour closer to the UI - providing the same defaults as the UI does

Before
----------------------------------------
v4 api requires fields for which reasonable defaults could be used. Both apis apply the 'wrong' default for contribution_status_id and as it is not required it is easy to mistakenly create recurring contributions that bypass the code logic to manage status

After
----------------------------------------
Defaults in line with the UI

Technical Details
----------------------------------------
This bubbled to the top because of pain around tests for apiv4

Comments
----------------------------------------
The change of status for contribution_status_id was previously discussed here https://github.com/civicrm/civicrm-core/pull/19512 - the key thing to note (which was obviously not clear up front) is that the Contribution BAO has code to update the recurring contribution status as contributions are created with contribution_recur_id set

### CODE REVIEW

File: CRM/Upgrade/Incremental/php/FiveThirtySeven.php [hunk 1]
```diff
@@ -106,4 +106,33 @@ public static function alterSavedSearchFK(CRM_Queue_TaskContext $ctx) {
...
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public static function updateDBDefaultsForContributionRecur(CRM_Queue_TaskContext $ctx): bool {
+    $pendingID = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_ContributionRecur', 'contribution_status_id', 'Pending');
+    CRM_Core_DAO::executeQuery("
+      ALTER TABLE `civicrm_contribution_recur`
+      MODIFY COLUMN `start_date` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'The date the first scheduled recurring contribution occurs.',
```
Comment (seamuslee001) [inline CRM/Upgrade/Incremental/php/FiveThirtySeven.php:129]: Should these date columns be datetime or timestamps?

Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/php/FiveThirtySeven.php:129]: @seamuslee001 I tested & the default works with datetime - which is what they already are. I wanted to leave any type change out of scope

Comment (JoeMurray) [reply CRM/Upgrade/Incremental/php/FiveThirtySeven.php:129]: I too prefer to have separate PR/issue for changing default value compared to changing the type.

File: CRM/Contribute/DAO/ContributionRecur.php [hunk 2]
```diff
@@ -587,7 +591,7 @@ public static function &fields() {
           'import' => TRUE,
           'where' => 'civicrm_contribution_recur.contribution_status_id',
           'export' => TRUE,
-          'default' => '1',
+          'default' => '2',
```
Comment (JoeMurray) [inline CRM/Contribute/DAO/ContributionRecur.php:594]: Might be nice to have comment 
// ie Pending

File: CRM/Upgrade/Incremental/php/FiveThirtySeven.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/php/FiveThirtySeven.php:129]: @JoeMurray I did have the PR for that change (linked above) - I closed it to incorporate into this as I felt it had been canvassed & upgrade scripts don't play well as separate PRs

File: CRM/Contribute/DAO/ContributionRecur.php (hunk 2 above)
Comment (colemanw) [reply CRM/Contribute/DAO/ContributionRecur.php:594]: Agree it would be nice, but we can't put comments in a generated DAO file.

**Comment by colemanw** [comment]
I'm merging as Eileen addressed feedback without any additional pushback, and I think this is better in than out.

**Comment by eileenmcnaughton** [comment]
Thanks @colemanw - one thing I've been mulling over is https://lab.civicrm.org/dev/core/-/issues/2474 & perhaps whether even just adding a try-catch might mitigate the risk of that

**Pull request merged**